client/dashboard/sites/site/environment-switcher.tsx

```diff
@@ -82,7 +83,25 @@ const EnvironmentSwitcherDropdown = ( {
 	}
 	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
 	const mutation = useMutation( stagingSiteCreateMutation( productionSite?.ID ?? 0 ) );
-	const handleCreate = () => {
+
+	const { data: connectionHealth } = useQuery( {
+		...jetpackConnectionHealthQuery( productionSite?.ID ?? 0 ),
+		enabled: !! productionSite?.ID,
+	} );
+
+	const handleCreate = async () => {
+		if ( ! connectionHealth?.is_healthy ) {
+			createErrorNotice(
+				__(
+					'You cannot create a staging site because your site has a Jetpack connection issue. Reload the page or contact support if it persists.'
```

Comment:
What if we updated the copy to something shorter like?

```suggestion
					'Unable to add a staging site due to a Jetpack connection issue. Reload the page or contact support if it persists.'
```

Reply:
Updated in https://github.com/Automattic/wp-calypso/pull/105648/commits/e37cc95ae80c92df84e2bbe0cb4fc3aedbd2fadb

Reply:
I used `cannot` instead of `unable` as it does not seem like we are using `unable` in the dashboard in general.

Reply:
That sounds good to me. 👍🏼 Thanks!